src/oaklib/implementations/semsimian/semsimian_implementation.py

```diff
@@ -56,8 +59,40 @@ def __post_init__(self):
             mn = m if isinstance(m, str) else m.__name__
             setattr(SemSimianImplementation, mn, methods[mn])
 
-        spo = [r for r in self.wrapped_adapter.relationships(include_entailed=True)]
-        self.semsimian = Semsimian(spo)
+        self.term_pairwise_similarity_attributes = [
+            attr
+            for attr in vars(TermPairwiseSimilarity)
+            if not any(attr.startswith(s) for s in ["class_", "_"])
+        ]
+
+    def _ensure_semsimian_object_is_set(
+        self, predicates: List[PRED_CURIE] = None, attributes: List[str] = None
```

Comment:
if the output changes based on the attribute list then the attribute list must be in the key - otherwise the cache could become corrupted (potentially silently)

but do you even need to parameterize by attributes? 

better to decide on a fixed set of attributes:

- jaccard
- IC of MRCA
- identifier(s) of MRCA(s)?

Reply:
This issue still remains as of now.

Reply:
If `attributes` is None, on the Semsimian side there already is a default list to fall back on. This list is:
```
            "subject_id",
            "object_id",
            "jaccard_similarity",
            "ancestor_information_content",
            "phenodigm_score",
            "cosine_similarity",
            "ancestor_id"
``` 

This is missing a few and I'm working on adding it.